### PR TITLE
Navigation UI

### PR DESCRIPTION
**Pull request by TheImpressiveDonut**

### CODE REVIEW

File: app/src/main/java/com/github/onedirection/navigation/fragment/account/AccountViewModel.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package com.github.onedirection.navigation.fragment.account;
+
+import androidx.lifecycle.ViewModel;
+
+public class AccountViewModel extends ViewModel {
+    // TODO: Implement the ViewModel
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/account/AccountViewModel.java:6]: There seems to be something to do here :)

File: app/src/main/java/com/github/onedirection/navigation/fragment/account/ExpandableListData.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        option2.add("Sub-option 2");
+
+        List<String> option3 = new ArrayList<String>();
+        option3.add("Sub-option 1");
+        option3.add("Sub-option 2");
+
+        expandableListDetail.put("Option 1", option1);
+        expandableListDetail.put("Option 2", option2);
+        expandableListDetail.put("Option 3", option3);
+        return expandableListDetail;
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/account/ExpandableListData.java:27]: Not sure we should leave those placeholder menus

File: app/src/main/java/com/github/onedirection/navigation/fragment/account/ui/login/LoginFragment.java [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                    return;
+                }
+                if (loginResult.getError() != null) {
+                    showLoginFailed(loginResult.getError());
+                }
+                if (loginResult.getSuccess() != null) {
+                    updateUiWithUser(loginResult.getSuccess());
+                }
+            }
+        });
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/account/ui/login/LoginFragment.java:76]: I guess this should be replaced with our login logic; don't know if you intend to do this in a future PR or if this is a oversight

File: app/src/main/java/com/github/onedirection/navigation/fragment/account/ui/login/LoginResult.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    @Nullable
+    LoggedInUserView getSuccess() {
+        return success;
+    }
+
+    @Nullable
+    Integer getError() {
+        return error;
+    }
+}
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/account/ui/login/LoginResult.java:31]: This class is most likely useless in our model (we use a CompletableFuture<User> instead)

**Comment by Ef55** [comment]
Final comment and I stop being annoying: be careful of Lint warnings !

**Review by Ef55 — APPROVED** [review]

**Review by Sosol02 — APPROVED** [review]

File: app/src/main/java/com/github/onedirection/navigation/fragment/account/ui/login/LoginResult.java (hunk 4 above)
Comment (TheImpressiveDonut) [reply app/src/main/java/com/github/onedirection/navigation/fragment/account/ui/login/LoginResult.java]: I removed it

File: app/src/main/java/com/github/onedirection/navigation/fragment/account/ui/login/LoginFragment.java (hunk 3 above)
Comment (TheImpressiveDonut) [reply app/src/main/java/com/github/onedirection/navigation/fragment/account/ui/login/LoginFragment.java]: Done

File: app/src/main/java/com/github/onedirection/navigation/fragment/account/AccountViewModel.java (hunk 1 above)
Comment (TheImpressiveDonut) [reply app/src/main/java/com/github/onedirection/navigation/fragment/account/AccountViewModel.java]: It was the automatic template of android studio, I removed it

**Pull request merged**